pkg/rest/retry.go

```diff
@@ -77,6 +77,10 @@ func GetWithRetry(client *http.Client, url string, settings RetrySetting) (resp
 
 // SendWithRetry will retry to call sendWithBody for a given number of times, waiting a give duration between calls
 func SendWithRetry(client *http.Client, sendWithBody SendRequestWithBody, objectName string, path string, body []byte, setting RetrySetting) (resp Response, err error) {
+	resp, err = sendWithBody(client, path, body)
```

Comment:
IIRC the Config Client calls to this happen after a first try failed (and a special retry case occurred)
While the Settings client goes straight to send with retry. 

Just changing this without modifying the handling for Config client errors, means we do an extra send/retry in their case right? 
I think that's visible from your change to the request_test assert as well, which expects +1 call. 

I don't think we should just do that, it kinda breaks the max retries by now always doing +1 "retry" call for Configs as they had an original API call before.